src/SearchField/index.scss

```diff
@@ -122,6 +122,12 @@
         background: map-get($search-btn-variants, "dark");
       }
     }
+
+    &:hover {
+      .btn-brand {
+        background: map-get($search-btn-variants, "dark");
```

Comment:
I don't think this fixes the issue, the color still does not change on hover for the `dark` variant
<img width="515" alt="image" src="https://github.com/openedx/paragon/assets/52399399/b9b12cd6-9cd3-4493-a2a0-f6aee0ddb96f">

it should change to this color (taken from the Buttons page of the docs site)
<img width="748" alt="image" src="https://github.com/openedx/paragon/assets/52399399/c1126d2d-b833-4e54-9728-673a8d0ac37c">